### ISSUE TITLE
Refine the Save modal design in site view

### ISSUE THREAD

**Issue by jameskoster** [WordPress/gutenberg#49832]
<img width="666" alt="Screenshot 2023-05-18 at 11 59 07" src="https://github.com/WordPress/gutenberg/assets/846565/a640ebe1-a0d2-4789-ab72-90f5b5776d98">

* Makes use of regular modal patterns (title, close button, bottom-actions).
* Actions are sticky at the bottom of the modal. This may require a component-level update. 
* Current context highlighted above all other changes.

<details><summary>Original issue</summary>In the lead up to the 6.2 release a global save button was added to the site editor. Clicking it invokes the following modal:

<img width="708" alt="Screenshot 2023-04-14 at 14 23 39" src="https://user-images.githubusercontent.com/846565/232055796-9d8a35f0-bf13-42b0-8082-f4327c9622a5.png">

This is a essentially a clone of the multi-entity save panel, presented in a modal. It needs to retain the same functionality of course, but the design needs some refinement. 

If we stick with the modal approach then it seems reasonable to at least use conventional anatomy (title, close button, actions at the bottom, etc), and be more accessible (https://github.com/WordPress/gutenberg/issues/47885). There might be different formats to explore for the contents as well.

I don't know how committed we are to the modal, it could also be worth exploring other options like drilldowns. </details>

**Comment by jasmussen** [comment]
> I don't know how committed we are to the modal, it could also be worth exploring other options like drilldowns.

The modal dialog has the benefit that it can be opened both from the save hub in the site view, and from the publish button in the edit view, and be the same interface. 

One added gotcha to consider, if not design for, is revisions. We won't need them for 6.3, but at some point you should be able to make a long sequence of changes (change site title, logo, tagline, style, templates) and then schedule them to go live on a particular date, ideally even with a title for the incoming change, such as "Black Friday design".

**Comment by paaljoachim** [comment]
It would be great to explore modal and drill down, so that we can look at advantages and disadvantages of both methods.
By doing so one would also gain a deeper knowledge for both methods.

**Comment by paaljoachim** [comment]
I want to add in this older issue.
**Improving multi-entity saving UI method.**
https://github.com/WordPress/gutenberg/issues/31456

**Comment by SaxonF** [comment]
Quick change here is to just use current modal patterns with actions on bottom. 

One other suggestion, which ties into the save hub work, is promoting the current context to be the first thing listed in the modal. e.g. if on `pages/my-page` than any changes to that page will be highlighted.


![Image](https://github.com/WordPress/gutenberg/assets/1072756/516ebbc3-6cd8-44dc-a8d6-39b15a1adafd)

**Comment by jasmussen** [comment]
Outside of some of the detail panel aspects in the background under the scrim, which are still works in progress, that modal design feels pretty good to me. Seems like we can update the main issue with this, what do you say?

**Comment by SaxonF** [comment]
@jasmussen yep fine with me. I don't think we currently have the ability to list out last modified time for each entity so can leave that off. We may also need to adapt modal to handle sticky footer unless it already handles that scenario. I'll update GH description tomorrow.

**Comment by paaljoachim** [comment]
Got a quick question.
In general for modals is it possible to adjust the screen locations of where these show up to keep these closer to where they were activated? As it would be helpful to associate the initial click with the result.

**Comment by jameskoster** [comment]
I tweaked the modal footer dimensions to be consistent with other modals and updated the OP.

**Comment by jasmussen** [comment]
Based on #57470 I did a little [quick work](https://www.figma.com/file/46qRP607dtqHGEosEaWWhY/Product-Design-Previously-6.4-Design?type=design&node-id=1424%3A1143&mode=design&t=iZ35UhHR40yxc9ep-1) to refine this a bit. One of the pieces to consider is multi entity saving as an interface you see also when switching themes, saving drafts, scheduling, etc. To that end, it may be worth not only having the single modal across site and edit views as already proposed, but expand the range of content that can be checked/unchecked. For example, you might want to switch to a new theme, but keep your colors and fonts, etc.

To that end, here are checkboxes and a list-view-esque expand/collapse for each section of content you can save:

<img width="2944" alt="Global Style changes, even longer term" src="https://github.com/WordPress/gutenberg/assets/1204802/a3bd5ef8-58de-47a6-b327-94ca23b46cf9">

The style changes based on the above are fairly minimal, a darker subheading style, and no separator. We can keep the "Current page" separation if that's a strong aspect of this.

Extrapolated from that is also here a nearer term version, that might be a stepping stone in implementation:

<img width="1536" alt="Global Style changes, longer term" src="https://github.com/WordPress/gutenberg/assets/1204802/a175ad84-d005-468b-bb2c-e9ce315fe2e1">

**Comment by SaxonF** [comment]
There is an opportunity for this work to think through how post/page publishing may tie into the save modal. If a user modifies a post, and only the post, we can put far greater emphasis on publishing as the core action of the modal so you still get that sense of accomplishment.

**Comment by jasmussen** [comment]
Noticed today that for the existing Save modal, if you click in the padding of the modal, it closes it. The area highlighted in green below.

![Screenshot 2024-08-27 at 11 37 23](https://github.com/user-attachments/assets/b5fb5b0e-6310-4a2b-9b37-fd5238398ce1)

I assume this will be solved by moving this save modal to a proper modal component, but tracking here as something to capture.

**Comment by afercia** [comment]
Chming-in here from https://github.com/WordPress/gutenberg/issues/67313 thanks to @jameskoster for pointing me at this issue.

Basically, the _content_ of this modal dialog is the `EntitiesSavedStates` component that is also used in the 'Save' panel. In the screenshot below:

- The site editor Save modal dialog.
- The Theme preview 'Activate theme` modal dialog.
- The Save panel.

![Image](https://github.com/user-attachments/assets/e0b732cf-5564-44b0-a743-6186012f4ac8)

While the content structure of the `EntitiesSavedStates` component makes sense in the Save panel, it's less than ideal when used inside a modal dialog:

- The order of the content isn't ideal.
- As far as I can tell, this is the only case where a modal dialog renders the Cancel / Save buttons at the top as first thing. They should be the last thing, right aligned.
- Any modal dialog should have a visible title at the top.
- The labeling of the dialog may be improved. Current example: `role="dialog" aria-label="Save site, content, and template changes"`.

As such, this is also an accessibility issue and I'd like to see some improvements here.
I'll try to work on small improvements here. Maybe not a full redesign as proposed in this issue but at least the order of the content and labeling should be fixed. It doesn't make much sense to just 'transplant' the `EntitiesSavedStates` content within this modal dialog because the content structure and semantics don't fit in the context of a modal dialog.

Related: https://github.com/WordPress/gutenberg/issues/67354

**Comment by afercia** [comment]
> Noticed today that for the existing Save modal, if you click in the padding of the modal, it closes it

For history: that's been fixed in https://github.com/WordPress/gutenberg/issues/67313 / https://github.com/WordPress/gutenberg/pull/67351
